src/ui/CategoryFacet/CategoryFacetSearch.ts

```diff
@@ -1,3 +1,199 @@
+  }
+
+  private handleFacetSearchFocus() {
+    if (this.currentlyDisplayedResults == null) {
+      this.displayNewValues();
+    }
+  }
+
+  private handleKeyboardEvent(event: KeyboardEvent) {
+    switch (event.which) {
+      case KEYBOARD.ENTER:
+        this.keyboardEventEnter();
+        break;
+      case KEYBOARD.DOWN_ARROW:
+        this.moveCurrentResultDown();
+        break;
+      case KEYBOARD.UP_ARROW:
+        this.moveCurrentResultUp();
+        break;
+      case KEYBOARD.ESCAPE:
+        this.closeSearchInput();
+        break;
+      default:
+        this.displayNewValues();
+    }
+  }
+
+  private moveCurrentResultDown() {
+    let nextResult = this.currentResult.el.nextSibling;
+    if (!nextResult) {
+      nextResult = first(this.facetSearchElement.searchResults.children);
+    }
+    this.setAsCurrentResult($$(<HTMLElement>nextResult));
+  }
+
+  private moveCurrentResultUp() {
+    let previousResult = this.currentResult.el.previousSibling;
+    if (!previousResult) {
+      previousResult = last(this.facetSearchElement.searchResults.children);
+    }
+    this.setAsCurrentResult($$(<HTMLElement>previousResult));
+  }
+
+  private keyboardEventEnter() {
+    this.selectCurrentResult();
+  }
+
+  private selectCurrentResult() {
+    if (this.currentResult) {
+      this.categoryFacet.changeActivePath(this.currentResult.el.dataset.path.split('|'));
+    }
+  }
+
+  private setAsCurrentResult(toSet: Dom) {
+    this.currentResult && this.currentResult.removeClass('coveo-category-facet-search-current-value');
+    this.currentResult = toSet;
+    toSet.addClass('coveo-category-facet-search-current-value');
+  }
+
+  private handleClickElsewhere(e: MouseEvent) {
+    if (!$$(<HTMLElement>e.target).closest('.coveo-category-facet-search-container')) {
+      this.closeSearchInput();
+    }
+  }
+
+  private closeSearchInput() {
+    this.withFacetSearchResults();
+    $$(this.categoryFacet.element).removeClass('coveo-category-facet-searching');
+    $$(this.facetSearchElement.searchResults).empty();
+    this.facetSearchElement.clearSearchInput();
+    this.facetSearchElement.hideSearchResultsElement();
+  }
+
+  private buildfacetSearchPlaceholder() {
+    const placeholder = $$('div', { className: 'coveo-category-facet-search-placeholder' });
+    placeholder.on('click', () => {
+      $$(this.categoryFacet.element).addClass('coveo-category-facet-searching');
+      this.focus();
+    });
+    const icon = $$('div', { className: 'coveo-category-facet-search-icon' }, SVGIcons.icons.checkboxHookExclusionMore);
+    SVGDom.addClassToSVGInContainer(icon.el, 'coveo-category-facet-search-icon-svg');
+
+    const label = $$('span', { className: 'coveo-category-facet-search-label' }, l('Search'));
+
+    placeholder.append(icon.el);
+    placeholder.append(label.el);
+    return placeholder;
+  }
+
+  private buildFacetSearchValue(categoryFacetValue: IGroupByValue) {
+    const path = categoryFacetValue.value.split('|');
```

Comment:
Many customers use a customer character to split their value in the old facet (option: delimitingCharacter). 

I just realized we are not currently allowing that anywhere with the new component.

Oh well... :/